tests/testthat/test-docnames.R

```diff
@@ -0,0 +1,12 @@
+context("test docnames")
+
+test_that("docnames always return names even if there aren't", {
+    
+    corp <- corpus(c('aaa', 'bbb', 'ccc'))
+    expect_equal(length(docnames(corp)), ndoc(corp))
+    
+    toks <- as.tokens(list('aaa', 'bbb', 'ccc'))
```

Comment:
I think that definitionally, we should never have `corpus` objects, and certainly never `tokens` objects, with no docnames. A better method would be to define them as an S4 object with a validator that requires docnames to be non-NULL and unique.